internal/impl/mysql/input_mysql_stream.go

```diff
@@ -0,0 +1,637 @@
+	msgChan               chan asyncMessage
+	batching              service.BatchPolicy
+	batchPolicy           *service.Batcher
+	tablesFilterMap       map[string]bool
+	checkPointLimit       int
+
+	logger *service.Logger
+	res    *service.Resources
+
+	streamCtx context.Context
+	errors    chan error
+	cp        *checkpoint.Capped[*int64]
+
+	snapshot                  *Snapshot
+	shutSig                   *shutdown.Signaller
+	snapshotMaxParallelTables int
+	fieldSnapshotMaxBatchSize int
+}
+
+const binLogCacheKey = "mysql_binlog_position"
+
+func newMySQLStreamInput(conf *service.ParsedConfig, res *service.Resources) (s service.BatchInput, err error) {
+	streamInput := mysqlStreamInput{
+		logger:                res.Logger(),
+		rawMessageEvents:      make(chan MessageEvent),
+		snapshotMessageEvents: make(chan MessageEvent),
+		msgChan:               make(chan asyncMessage),
+		res:                   res,
+		streamCtx:             context.Background(),
+
+		errors:  make(chan error, 1),
+		shutSig: shutdown.NewSignaller(),
+	}
+
+	var batching service.BatchPolicy
+
+	if streamInput.dsn, err = conf.FieldString(fieldMySQLDSN); err != nil {
+		return nil, err
+	}
+
+	streamInput.mysqlConfig, err = mysql.ParseDSN(streamInput.dsn)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing mysql DSN: %v", err)
+	}
+
+	if streamInput.tables, err = conf.FieldStringList(fieldMySQLTables); err != nil {
+		return nil, err
+	}
+
+	if streamInput.streamSnapshot, err = conf.FieldBool(fieldStreamSnapshot); err != nil {
+		return nil, err
+	}
+
+	if streamInput.snapshotMaxParallelTables, err = conf.FieldInt(fieldMaxSnapshotParallelTables); err != nil {
+		return nil, err
+	}
+
+	if streamInput.fieldSnapshotMaxBatchSize, err = conf.FieldInt(fieldSnapshotMaxBatchSize); err != nil {
+		return nil, err
+	}
+
+	if streamInput.checkPointLimit, err = conf.FieldInt(fieldCheckpointLimit); err != nil {
+		return nil, err
+	}
+
+	if streamInput.binLogCache, err = conf.FieldString(fieldCheckpointKey); err != nil {
+		return nil, err
+	}
+
+	i := &streamInput
+	i.cp = checkpoint.NewCapped[*int64](int64(i.checkPointLimit))
+
+	i.tablesFilterMap = map[string]bool{}
+	for _, table := range i.tables {
+		if err = validateTableName(table); err != nil {
+			return nil, err
+		}
+		i.tablesFilterMap[table] = true
+	}
+
+	res.Logger().Info("Starting MySQL stream input")
+
+	if batching, err = conf.FieldBatchPolicy(fieldBatching); err != nil {
+		return nil, err
+	} else if batching.IsNoop() {
+		batching.Count = 1
+	}
+
+	i.batching = batching
+	if i.batchPolicy, err = i.batching.NewBatcher(res); err != nil {
+		return nil, err
+	} else if batching.IsNoop() {
+		batching.Count = 1
+	}
+
+	r, err := service.AutoRetryNacksBatchedToggled(conf, i)
+	if err != nil {
+		return nil, err
+	}
+
+	return conf.WrapBatchInputExtractTracingSpanMapping("mysql_stream", r)
+}
+
+func init() {
+	err := service.RegisterBatchInput(
+		"mysql_stream", mysqlStreamConfigSpec,
```

Comment:
Based on some internal conversations, we should rename this to be more explicit it's a CDC connector

```suggestion
		"mysql_cdc", mysqlStreamConfigSpec,
```